Automattic/wp-calypso
# Notifications: Rewrite AuthWrapper to use React Hooks internally

#### Changes proposed in this Pull Request

* Re-write the `AuthWrapper` component in `apps/notifications/src/standalone` so that it uses React Hooks (as a possible first step toward phasing out the higher-order component in favor of a hook).
* Separate the two available authentication methods -- OAuth and the iframe-based proxy -- into their own components, to simplify the overall logic.

#### Testing instructions

* In a sandbox environment, build the notifications app (`cd apps/notifications && yarn build`) and deploy it to `widgets.wp.com/notifications` according to the instructions in [`README.md`](https://github.com/Automattic/wp-calypso/blob/5a2afd36bdd284106e2fe56967d0a72ef976ec7d/apps/notifications/README.md).
* Add a DNS alias in your `hosts` file so that requests to `widgets.wp.com` load from your sandbox environment.
* Verify that the notifications panel still loads and behaves correctly in the masterbar for the following experiences and site types:
  * inside Calypso;
  * WordPress.com simple sites;
  * WordPress.com eCommerce ("Atomic") sites; and
  * self-hosted Jetpack sites.
